### PR TITLE
feat: Implement hint on uint384_extension lib

### PR DESCRIPTION
**Pull request by fmoletta**
Depends on #960 
Implements the following hint: 
```
           def split(num: int, num_bits_shift: int, length: int):
               a = []
               for _ in range(length):
                   a.append( num & ((1 << num_bits_shift) - 1) )
                   num = num >> num_bits_shift
               return tuple(a)

           def pack(z, num_bits_shift: int) -> int:
               limbs = (z.d0, z.d1, z.d2)
               return sum(limb << (num_bits_shift * i) for i, limb in enumerate(limbs))

           def pack_extended(z, num_bits_shift: int) -> int:
               limbs = (z.d0, z.d1, z.d2, z.d3, z.d4, z.d5)
               return sum(limb << (num_bits_shift * i) for i, limb in enumerate(limbs))

           a = pack_extended(ids.a, num_bits_shift = 128)
           div = pack(ids.div, num_bits_shift = 128)

           quotient, remainder = divmod(a, div)

           quotient_split = split(quotient, num_bits_shift=128, length=6)

           ids.quotient.d0 = quotient_split[0]
           ids.quotient.d1 = quotient_split[1]
           ids.quotient.d2 = quotient_split[2]
           ids.quotient.d3 = quotient_split[3]
           ids.quotient.d4 = quotient_split[4]
           ids.quotient.d5 = quotient_split[5]

           remainder_split = split(remainder, num_bits_shift=128, length=3)
           ids.remainder.d0 = remainder_split[0]
           ids.remainder.d1 = remainder_split[1]
           ids.remainder.d2 = remainder_split[2]
```

## Checklist
- [ ] Linked to Github Issue
- [x] Unit tests added
- [x] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
  - [ ] CHANGELOG has been updated.

### CODE REVIEW

File: src/hint_processor/builtin_hint_processor/uint384_extension.rs [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    let a = pack_extended(Uint768::from_var_name("a", vm, ids_data, ap_tracking)?, 128);
+    let div = pack(
+        BigInt3::from_var_name("div", vm, ids_data, ap_tracking)?,
+        128,
+    );
+    let quotient_addr = get_relocatable_from_var_name("quotient", vm, ids_data, ap_tracking)?;
+    let remainder_addr = get_relocatable_from_var_name("remainder", vm, ids_data, ap_tracking)?;
+    let (quotient, remainder) = a.div_mod_floor(&div);
```
Comment (Oppen) [inline src/hint_processor/builtin_hint_processor/uint384_extension.rs:127]: Is there any chance `div` is `0`? If there is, we should catch it, I believe.

**Review by Oppen — COMMENTED** [review]
LGTM, just one comment.

**Review by pefontana — APPROVED** [review]

File: src/hint_processor/builtin_hint_processor/uint384_extension.rs (hunk 1 above)
Comment (fmoletta) [reply src/hint_processor/builtin_hint_processor/uint384_extension.rs:127]: Added check to hint & previous uint384 hints with similar behaviour

Comment (fmoletta) [reply src/hint_processor/builtin_hint_processor/uint384_extension.rs:127]: Edit: I added it to another PR that depends on this, let me copy it to this branch

**Review by Oppen — APPROVED** [review]

**Comment by Oppen** [comment]
There seems to be a significant, reproducible (I retried the workflow twice) slowdown in several benchmarks. I think we should figure out what causes it before merging.
I'll try to run the benchmarks by hand in the server tomorrow, if we're lucky it's just that by chance the memory layout became less efficient in this particular build. If we're not, it may be a fun ride.

**Pull request merged**